### ISSUE TITLE
Error on Server Startup

### ISSUE THREAD

**Issue by TechnicJelle** [Tomson124/SimplyJetpacks2#32]
I'm getting an error when I start my modded server:
```
22:02:46 [Worker-Main-12/ERROR] [minecraft/RecipeManager]: Parsing error loading recipe simplyjetpacks:sj/particle_rainbow
22:02:46 com.google.gson.JsonSyntaxException: Unknown item tag 'forge:dusts/coal'
22:02:46 	at net.minecraft.item.crafting.Ingredient.func_199803_a(Ingredient.java:223) ~[?:?] {re:classloading,pl:accesstransformer:B}
22:02:46 	at net.minecraftforge.common.crafting.VanillaIngredientSerializer.parse(VanillaIngredientSerializer.java:41) ~[forge:?] {re:classloading}
22:02:46 	at net.minecraftforge.common.crafting.CraftingHelper.getIngredient(CraftingHelper.java:155) ~[forge:?] {re:classloading}
22:02:46 	at net.minecraft.item.crafting.Ingredient.func_199802_a(Ingredient.java:189) ~[?:?] {re:classloading,pl:accesstransformer:B}
22:02:46 	at net.minecraft.item.crafting.ShapedRecipe.func_192408_a(ShapedRecipe.java:252) ~[?:?] {re:classloading,pl:accesstransformer:B}
22:02:46 	at net.minecraft.item.crafting.ShapedRecipe.access$000(ShapedRecipe.java:24) ~[?:?] {re:classloading,pl:accesstransformer:B}
22:02:46 	at net.minecraft.item.crafting.ShapedRecipe$Serializer.func_199425_a_(ShapedRecipe.java:276) ~[?:?] {re:classloading}
22:02:46 	at net.minecraft.item.crafting.ShapedRecipe$Serializer.func_199425_a_(ShapedRecipe.java:272) ~[?:?] {re:classloading}
22:02:46 	at net.minecraft.item.crafting.RecipeManager.func_215377_a(RecipeManager.java:141) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:immersiveengineering.mixins.json:accessors.RecipeManagerAccess,pl:mixin:APP:performant.mixins.json:item.RecipeManagerMixin,pl:mixin:A,pl:runtimedistcleaner:A}
22:02:46 	at net.minecraft.item.crafting.RecipeManager.func_212853_a_(RecipeManager.java:61) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:immersiveengineering.mixins.json:accessors.RecipeManagerAccess,pl:mixin:APP:performant.mixins.json:item.RecipeManagerMixin,pl:mixin:A,pl:runtimedistcleaner:A}
22:02:46 	at net.minecraft.item.crafting.RecipeManager.func_212853_a_(RecipeManager.java:38) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:immersiveengineering.mixins.json:accessors.RecipeManagerAccess,pl:mixin:APP:performant.mixins.json:item.RecipeManagerMixin,pl:mixin:A,pl:runtimedistcleaner:A}
22:02:46 	at net.minecraft.client.resources.ReloadListener.func_215269_a(SourceFile:13) ~[?:?] {re:classloading,re:mixin}
22:02:46 	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:714) ~[?:?] {}
22:02:46 	at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:478) ~[?:?] {}
22:02:46 	at net.minecraft.resources.AsyncReloader.func_219557_a(SourceFile:71) ~[?:?] {re:classloading}
22:02:46 	at net.minecraft.resources.AsyncReloader.func_219560_a(SourceFile:70) ~[?:?] {re:classloading}
22:02:46 	at java.util.concurrent.CompletableFuture$UniCompletion.claim(CompletableFuture.java:568) ~[?:?] {}
22:02:46 	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:710) ~[?:?] {}
22:02:46 	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506) ~[?:?] {}
22:02:46 	at java.util.concurrent.CompletableFuture.postFire(CompletableFuture.java:610) ~[?:?] {}
22:02:46 	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:722) ~[?:?] {}
22:02:46 	at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:478) ~[?:?] {}
22:02:47 	at net.minecraft.resources.AsyncReloader.func_219557_a(SourceFile:71) ~[?:?] {re:classloading}
22:02:47 	at net.minecraft.resources.AsyncReloader.func_219560_a(SourceFile:70) ~[?:?] {re:classloading}
22:02:47 	at java.util.concurrent.CompletableFuture$UniCompletion.claim(CompletableFuture.java:568) ~[?:?] {}
22:02:47 	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:710) ~[?:?] {}
22:02:47 	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506) ~[?:?] {}
22:02:47 	at java.util.concurrent.CompletableFuture.postFire(CompletableFuture.java:610) ~[?:?] {}
22:02:47 	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:722) ~[?:?] {}
22:02:47 	at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:478) ~[?:?] {}
22:02:47 	at net.minecraft.resources.AsyncReloader.func_219557_a(SourceFile:71) ~[?:?] {re:classloading}
22:02:47 	at net.minecraft.resources.AsyncReloader.func_219560_a(SourceFile:70) ~[?:?] {re:classloading}
22:02:47 	at java.util.concurrent.CompletableFuture$UniCompletion.claim(CompletableFuture.java:568) ~[?:?] {}
22:02:47 	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:710) ~[?:?] {}
22:02:47 	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506) ~[?:?] {}
22:02:47 	at java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2073) ~[?:?] {}
22:02:47 	at net.minecraft.resources.AsyncReloader$1.func_216873_a(SourceFile:53) ~[?:?] {re:classloading}
22:02:47 	at net.minecraft.resources.AsyncReloader$1.func_216872_a(SourceFile:50) ~[?:?] {re:classloading}
22:02:47 	at java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:1072) [?:?] {}
22:02:47 	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506) [?:?] {}
22:02:47 	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1705) [?:?] {}
22:02:47 	at java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1692) [?:?] {}
22:02:47 	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290) [?:?] {}
22:02:47 	at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020) [?:?] {}
22:02:47 	at java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656) [?:?] {}
22:02:47 	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594) [?:?] {}
22:02:47 	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183) [?:?] {}

```

This happens for all the particle recipes: 
- simplyjetpacks:sj/particle_rainbow
- simplyjetpacks:sj/particle_smoke
- simplyjetpacks:sj/particle_none
- simplyjetpacks:sj/particle_default

It doesn't really surprise me that it can't find the dust, because there indeed is no coal dust in my modpack, but I didn't expect it to be a problem. Would it be possible to catch this error and in that case replace the coal dust item with some comparable vanilla item? Maybe a torch? Or black dye? Redstone dust?

**Comment by Tomson124** [comment]
Yes it would, could you please specify which MC version you are on? And maybe provide a list of mods, maybe there is somewhere an alternative that I can implement, too.

**Comment by TechnicJelle** [comment]
My server is on 1.16.5 using forge forge-1.16.5-36.0.1

<details>
<summary>Modlist</summary>

- absentbydesign-1.16.4-1.3.1.jar
- AI-Improvements-1.16.2-0.3.0.jar
- AppleSkin-mc1.16.2-forge-1.0.14.jar
- appliedenergistics2-8.1.0.jar
- architectury-1.4.98-forge.jar
- AutoRegLib-1.6-47.jar
- BetterAdvancements-1.16.4-0.1.0.104.jar
- BlueMap-1.3.1-forge-1.16.4.jar
- Bookshelf-1.16.5-10.0.1.jar
- BowInfinityFix-1.16.x-rv9.1.jar
- bwncr-1.16.4-3.9.16.jar
- ChineseWorkshop-1.16.3-3.2.0.jar
- chiselsandbits-0.2.10-RELEASE.jar
- chocolate-1.3.0-1.16.4.jar
- Chunk+Pregenerator-V1.16-3.0.3.jar.disabled
- cloth-config-forge-4.1.3.jar
- Clumps-6.0.0.13.jar
- CodeChickenLib-1.16.4-3.5.1.408-universal.jar
- comforts-forge-1.16.4-4.0.1.0.jar
- Controlling-7.0.0.12.jar
- CosmeticArmorReworked-1.16.5-v1.jar
- CraftingTweaks_1.16.3-12.2.0.jar
- create-mc1.16.3_v0.3e.jar
- createplus-1.16.4_v0.3.jar
- dcintegration-2.1.0-1.16.jar
- decorative_blocks-1.16.4-1.7.2.jar
- DefaultOptions_1.16.3-12.2.0.jar
- DrawersTooltip-1.16.2-2.1.0.jar
- EnchantmentDescriptions-1.16.4-6.0.3.jar
- EnderStorage-1.16.4-2.7.1.166-universal.jar
- endertanks-1.16-1.9.3.jar
- engineersdecor-1.16.4-1.1.7.jar
- extremeSoundMuffler-3.6_Forge-1.16.4.jar
- FastFurnace-1.16.4-4.4.0.jar
- FastLeafDecay-v25.jar
- FastWorkbench-1.16.4-4.5.0.jar
- framedcompactdrawers-1.16-2.2.0.jar
- ftb-backups-2.1.1.6.jar
- ftb-chunks-1604.2.2.61.jar
- ftb-gui-library-1604.2.0.30-forge.jar
- glassential-forge-1.16.4-1.1.5.jar
- gravestone-1.16.5-1.0.0.jar
- ImmersiveEngineering-1.16.5-4.2.0-130.jar
- ImmersivePetroleum-1.16.4-3.1.0-2.jar
- immersiveposts-1.16.3-3.0.0.jar
- InfinityMending-1.16.3-0.1.0.jar
- itemzoom-1.16.4-2.3.0.jar
- jei-1.16.4-7.6.1.65.jar
- jei-professions-1.0.0-1.16.4.jar
- jeiintegration_1.16.4-6.1.1.11.jar
- JustEnoughResources-1.16.4-0.12.0.104.jar
- Kiwi-1.16.3-3.4.1.jar
- LastPlayedLogger-1.16.5-1.0.0.jar.disabled
- light-overlay-5.6.1.jar
- mcw-doors-1.0.1fix-mc1.16.4.jar
- mcw-furniture-2.0.0-mc1.16.4.jar
- mcw-paintings-1.0.0-mc1.16.4-5.jar
- mcw-trapdors-1.0.0-mc1.16.4.jar
- mcw-windows-1.0.2-mc1.16.4.jar
- mcws-roofs-2.0.0-mc1.16.4.jar
- MixinBootstrap-1.0.5.jar
- MouseTweaks-2.13-mc1.16.2.jar
- NetherPortalFix_1.16.3-7.2.1.jar
- overloadedarmorbar-5.1.0.jar
- Patchouli-1.16.4-48.jar
- performant-1.16.2-4-3.30m.jar
- pitg-1.16.3-1.4.1.jar
- Placebo-1.16.4-4.4.1.jar
- pneumaticcraft-repressurized-1.16.4-2.9.3-120.jar
- Quark-r2.4-297.jar
- shetiphiancore-1.16-3.8.4.jar
- SimplyJetpacks2-1.16.4-2.6.6.0.jar
- snowundertrees-1.16.4-v1.1.3.jar
- StorageDrawers-1.16.3-8.2.2.jar
- swingthroughgrass-1.16.4-1.5.3.jar
- theoneprobe-1.16-3.0.7.jar
- Toast-Control-1.16.4-4.3.1.jar
- voyage-1.16.4-1.0.3.jar
- windowlogging-mc1.16.3_v0.0.2.jar
- WorldPreGenerator-1.16.5-1.0.0.jar

</details>

I hope this helps!

**Comment by H0bbitBaron** [comment]
Same for me https://pastebin.com/yF0emvgK

**Comment by tomheaton** [comment]
Will be fixed in next 1.16.5 update 🦖
